src/GraphQL/Types/ResolveFieldContext.cs

```diff
@@ -106,6 +108,26 @@ public bool HasArgument(string argumentName)
         {
             return Arguments?.ContainsKey(argumentName) ?? false;
         }
+
+        public async Task<object> TryAsyncResolve(Func<ResolveFieldContext<TSource>, Task<object>> resolve, Func<ExecutionErrors, Task<object>> error = null)
+        {
+            try
+            {
+                return await resolve(this);
+            }
+            catch (Exception ex)
+            {
+                if (error == null)
+                {
+                    Errors.Add(new ExecutionError(ex.Message));
```

Comment:
Do we lose context if only exception message is added? Should it contain the whole ex.ToString()?

Reply:
Especially in case of AggregateException etc..

Reply:
Could perhaps add the full exception as the inner exception to keep the simpler top level message.

Reply:
@joemcbride the point of this method is to allow nicer messages. In my case for example I am throwing a contextual message to the user. I didn't include the stack trace / inner exception for that reason. I guess one option is to have a switch that lets you decide if you want the inner exception.

Reply:
The inner exception will never been seen in the response unless you have `ExposeExceptions` set on the `ExecutionOptions`.  I'm fine with either way.